src/adapters/deno.ts

```diff
@@ -35,10 +35,12 @@ const denoAdapter: Adapter<DenoAdapter, DenoOptions> = (options = {}) => {
       if (endResponse) {
         return endResponse;
       }
-
+      // prettier-ignore
+      const headers = upgradeHeaders instanceof Headers ? upgradeHeaders : new Headers(upgradeHeaders);
```

Comment:
It is little bit costy to create a new `Headers` object we could directly check 3 versions .